cpp/src/io/utilities/hostdevice_vector.hpp

```diff
@@ -37,71 +40,53 @@ class hostdevice_vector {
  public:
   using value_type = T;
 
-  hostdevice_vector() {}
+  hostdevice_vector() : hostdevice_vector(0, cudf::default_stream_value) {}
 
-  hostdevice_vector(hostdevice_vector&& v) { move(std::move(v)); }
-  hostdevice_vector& operator=(hostdevice_vector&& v)
-  {
-    move(std::move(v));
-    return *this;
-  }
-
-  explicit hostdevice_vector(size_t max_size, rmm::cuda_stream_view stream)
-    : hostdevice_vector(max_size, max_size, stream)
+  explicit hostdevice_vector(size_t size, rmm::cuda_stream_view stream)
+    : hostdevice_vector(size, size, stream)
   {
   }
 
   explicit hostdevice_vector(size_t initial_size, size_t max_size, rmm::cuda_stream_view stream)
-    : max_elements(max_size), num_elements(initial_size)
-  {
-    if (max_elements != 0) {
-      CUDF_CUDA_TRY(cudaMallocHost(reinterpret_cast<void**>(&h_data), sizeof(T) * max_elements));
-      d_data.resize(sizeof(T) * max_elements, stream);
-    }
-  }
-
-  ~hostdevice_vector()
+    : d_data(0, stream)
```

Comment:
why is this needed?

Reply:
device_uvector doesn't have a default constructor, so we need to provide a stream